src/client/Microsoft.Identity.Client/Cache/CacheSessionManager.cs

```diff
@@ -105,7 +105,7 @@ private async Task RefreshCacheForReadOperationsAsync(CacheEvent.TokenTypes cach
                     };
 
                     _requestParams.RequestContext.Logger.Verbose("[Cache Session Manager] Waiting for cache semaphore");
-                    await TokenCacheInternal.Semaphore.WaitAsync().ConfigureAwait(false);
+                    await TokenCacheInternal.Semaphore.WaitAsync(_requestParams.RequestContext.UserCancellationToken).ConfigureAwait(false);
```

Comment:
@jennyf19 - this is already in the drop we gave MyApps

Reply:
i think so.